index.js

```diff
@@ -0,0 +1,217 @@
+		logger.info(`Getting price`);
+		const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		const API_WORLD_URL = `browsequotes/v1.0/US/${this.config.currency}/en-US`;
+		const place = `${this.config.departurePlace}-sky/${this.config.destinationPlace}-sky`;
+		const date = `${this.config.departureDate}?inboundpartialdate=${this.config.returnDate}`;
+		const settings = {
+			url: `${API_BASE_URL}/${API_WORLD_URL}/${place}/${date}`,
+			method: 'GET',
+			headers: {
+				'x-rapidapi-host': 'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
+				'x-rapidapi-key': this.authorization.apiKey
+			}
+		}
+		if (!this.authorization.apiKey) {
+			throw 'Missing API key';
+		}
+		return request(settings).then(answer => {
+			const priceFromApi = JSON.parse(answer);
+			// If no prices are listed
+			// It will display `This flight is not listed. Please modify your request.` 
+			// See the logic line 118
+			if (priceFromApi.Quotes.length == 0) {
+				return null;
+			}
+			// Collects the first element of "Quotes" corresponding 
+			// to the minprice for the selected flight.
+			return priceFromApi.Quotes[0].MinPrice;
+		});
+	}
+
+	// This search input allows the user to find easily the name of an airport. 
+	// The user starts to write the name of a city or an airport he needs, then
+	// the search input will show him the different airports linked to his input.
+	// It is a search method linked to a JSON file that holds the IATA code and the name
+	// of the airports we need.
+	async options(fieldId, search) {
+		// const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		if (airports) {
+			logger.info("Sending preloaded airports");
+			return this.getIATA(airports, search);
+		} else if (fs.existsSync('./airports.json')) {
+			logger.info('Loading places from a file.');
+			const airports = require('./airports.json');
+			return this.getIATA(airports, search);
+		}
+	}
+
+	// Parameters needed to create the search method and collect the right datas: IATAs and names of
+	// the airports.
+	/**
+	* @param {*} airports
+	* @param {String} search 
+	*/
+	async getIATA(airports, search) {
+		if (search != null) {
+			search = search.trim().toLowerCase();
+		}
+		let option = [];
+		for (const airport of airports) {
+			// Datas needed: the user will enter the name of the airport (value) which is
+			// linked to the IATA (key).
+			// The IATA is needed cause it will be added in the API url to allow the
+			// collect of the price.
+			const key = airport.iata;
+			let value = airport.name || '';
+			if (!search || value.toLowerCase().includes(search)) {
+				option.push({
+					key: key,
+					value: value
+				});
+			}	
+		}
+		return option;
+	}	
+
+	// Store price obtained from first loaded.
+	// The price is updated every 24 hours.
+	// This price is compared with the price collected from the API every minute.
+	storeFirstPriceOfTheDay(price) {
```

Comment:
Almost. Because you don't always store it. You only store it if 24hours passed or nothing is stored.
I would say. storeFirstProceOfTheDayIfNeeded(). It's not perfect but it's better